scene/gui/file_dialog.cpp

```diff
@@ -87,6 +87,10 @@ void FileDialog::_notification(int p_what) {
 			if (!is_visible()) {
 				set_process_shortcut_input(false);
 			}
+
+			if (invalidated) {
+				invalidate(); // Put it here to preview in the editor.
+			}
```

Comment:
Just call it unconditionally, otherwise this is very confusing because of the chosen semantics. If you call it unconditionally, it will update the list on becoming visible and mark itself as invalidated on becoming invisible. Which seems fine and correct.